lab_1/main.py

```diff
@@ -86,10 +167,44 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
+    if not isinstance(unknown_profile, dict):
+        return None
+    if not isinstance(profile_to_compare, dict):
+        return None
+    if not isinstance(top_n, int):
+        return None
+
+    popular_words = get_top_n_words(profile_to_compare["freq"], top_n)
+    popular_words_un = get_top_n_words(unknown_profile["freq"], top_n)
+
+    top_common = []
+    for i in popular_words:
+        if i in popular_words_un:
+            top_common.append(i)
+
+    score = len(top_common) / len(popular_words_un)
+
+    new_keys = list((profile_to_compare["freq"]).keys())
+    len_of_words_dict = {}
+    for i in new_keys:
+        len_of_words_dict[len(i)] = i
```

Comment:
I can see a problem here

if there are tokens with the same length?
wouldn't keys change?

Reply:
Yes, they probably would. Will it change if I make a dict with words as keys and lengths as values?